File: php/145-binary-tree-postorder-traversal.php
<?php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($value) { $this->val = $value; }
 * }
 */
class Solution 
{
    private $data = [];
    /**
     * @param TreeNode $root
     * @return Integer[]
     */
    public function postorderTraversal($root) 
    {
        if ($root) {
            $this->postorderTraversal($root->left);
            $this->postorderTraversal($root->right);
            $this->data[] = $root->val;
        }
        return $this->data;
    }
    
}

File: go/145-binary-tree-postorder-traversal.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
	var res []int
	traversal(root, &res)
	return res
}
func traversal(root *TreeNode, res *[]int) {
	if root == nil {
		return
	}
	traversal(root.Left, res)
	traversal(root.Right, res)
	*res = append(*res, root.Val)
}
